### PR TITLE
boards/x86: up_squared: enable GPIO

### PR DESCRIPTION
**Pull request by dcpleung**
This PR adds a new GPIO driver for the Intel Apollo Lake SoC, and enables it the UP Squared board configuration. This also includes a sample app to utilize GPIOs on the UP Squared board.

### CODE REVIEW

File: dts/bindings/gpio/intel,apl-gpio.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+description: >
+    This is a representation of the Intel Apollo Lake GPIO node
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "intel,apl-gpio"
+
```
Comment (galak) [inline dts/bindings/gpio/intel,apl-gpio.yaml]: `generation: define` is missing

File: boards/x86/up_squared/up_squared.dts [hunk 2]
```diff
@@ -155,5 +155,21 @@
 
 			status = "ok";
 		};
+
+		gpio: gpio@0 {
```
Comment (galak) [inline boards/x86/up_squared/up_squared.dts]: Why is this in the board.dts vs apollo_lake.dtsi

**Review by galak — DISMISSED** [review]
Some dt changes needed.

**Review by dbkinder — APPROVED** [review]
works for me, thanks.

**Review by galak — APPROVED** [review]
DT parts look fine to me.

**Review by nashif — APPROVED** [review]

**Pull request merged**